pkg/minikube/cluster/cluster.go

```diff
@@ -618,3 +618,17 @@ func GetServiceURLs(api libmachine.API, namespace string, t *template.Template)
 
 	return serviceURLs, nil
 }
+
+//Create the host directories in the virtual machine if not present
+func EnsureHostDirectories(api libmachine.API, dirs []string) (error) {
```

Comment:
cluster.go is also a mess, I can see few functions which should not be part of it. I will try to fix some of it in this PR.